### PR TITLE
moved old script api to model.script

### PR DESCRIPTION
**Pull request by kaikreuzer**
I have removed the core.scriptengine bundle and moved the code into the model.script bundle as it is highly linked to the current Xbase rule engine.
This is a preparation for the introduction of the new rule engine. For this, I have decided that scripting support should make use of javax.script, which comes with very similar interfaces to what is currently found in core.scriptengine. In a future step, the Xbase script engine can be integrated through javax.script as well, so that it becomes available as an option in the new rule engine.

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
@maggu2810 May I ask you to review this change?

**Comment by maggu2810** [comment]
Sure, will add it to my todo list.

File: bundles/core/org.eclipse.smarthome.core.transform/OSGI-INF/transformationaction.xml [hunk 1]
```diff
@@ -12,6 +12,6 @@
 	<implementation class="org.eclipse.smarthome.core.transform.internal.TransformationActionService" />
 
 	<service>
-		<provide interface="org.eclipse.smarthome.core.scriptengine.action.ActionService" />
+		<provide interface="org.eclipse.smarthome.model.script.actions.ActionService" />
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.transform/OSGI-INF/transformationaction.xml]: Isn't the interface "org.eclipse.smarthome.model.script.engine.action.ActionService" now?

File: bundles/designer/org.eclipse.smarthome.designer.core/src/main/java/org/eclipse/smarthome/designer/core/config/PluginProjectCreator.xtend [hunk 2]
```diff
@@ -124,8 +125,12 @@ class PluginProjectCreator implements IProjectCreator {
 	}
 	
 	def private getImportedPackages() {
-		val actionServices = CoreActivator.actionServiceTracker.services.filter(ActionService)
-		actionServices.map[it.actionClass.package.name].toSet
+		val actionServices = CoreActivator.actionServiceTracker.services?.filter(ActionService)
+		if (actionServices!=null) {
```
Comment (maggu2810) [inline bundles/designer/org.eclipse.smarthome.designer.core/src/main/java/org/eclipse/smarthome/designer/core/config/PluginProjectCreator.xtend]: The formatter handles .java files only. Correct?
Perhaps we should add the spaces around != manually.
Yes, that is a little bit pedantic :)

**Comment by maggu2810** [comment]
The code seems okay for me.
What about runtime tests?

File: bundles/designer/org.eclipse.smarthome.designer.core/src/main/java/org/eclipse/smarthome/designer/core/config/PluginProjectCreator.xtend (hunk 2 above)
Comment (kaikreuzer) [inline bundles/designer/org.eclipse.smarthome.designer.core/src/main/java/org/eclipse/smarthome/designer/core/config/PluginProjectCreator.xtend]: fixed.

File: bundles/core/org.eclipse.smarthome.core.transform/OSGI-INF/transformationaction.xml (hunk 1 above)
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.transform/OSGI-INF/transformationaction.xml]: good catch - somehow the automatic refactoring has missed that.

**Comment by kaikreuzer** [comment]
> Added two comments, the remaining code seems okay for me.

Thanks, fixed them.

> What about runtime tests?

The existing tests should still work. As I only moved some code, I didn't add any new tests though.

**Comment by maggu2810** [comment]
> The existing tests should still work. As I only moved some code, I didn't add any new tests though.

I don't think about test cases, just something like
* start an ESH runtime
* look if all bundles are resolved
* look if all services are satisfied
* ...

Have I to do it?

**Comment by kaikreuzer** [comment]
I ran the mvn build and tested both launch configs (runtime+designer) from within the IDE.

> Have I to do it?

This is up to you. We usually only actively test things on larger PRs, where executing the code is part of the review (for discovering bugs) as well. If code is only moved around or small changes are done, the reviewer does not have to do such time consuming tests.

**Comment by kaikreuzer** [comment]
If ok for you, I would merge it then (did a rebase already)

**Comment by maggu2810** [comment]
Yes it is (ok for me).

**Pull request merged**